toolkit/cli/commands/src/address_association_signatures.rs

```diff
@@ -9,22 +9,54 @@ use serde_json::json;
 use sidechain_domain::*;
 use std::str::FromStr;
 
+/// Command structure for generating address association signatures.
+///
+/// This struct represents the parameters required to create cryptographic signatures
+/// that establish an association between a Cardano stake address and a Partner Chain address.
+/// Address associations enable cross-chain operations by linking identities between
+/// the mainchain (Cardano) and sidechain (Partner Chain) networks.
+///
+/// ## Address Association Process
+///
+/// The command generates an Ed25519 signature over a structured message that includes:
+/// - The Cardano stake public key (derived from the signing key)
+/// - The Partner Chain address to be associated
+/// - The genesis UTXO identifying the specific Partner Chain instance
+///
+/// This signature proves that the holder of the Cardano stake signing key authorizes
+/// the association with the specified Partner Chain address.
+///
+/// ## Generic Address Type
+///
+/// The struct is generic over `PartnerchainAddress` to support different address
+/// formats used by various Partner Chain implementations (e.g., AccountId32, H160).
 #[derive(Clone, Debug, Parser)]
 #[command(author, version, about, long_about = None)]
 pub struct AddressAssociationSignaturesCmd<
 	PartnerchainAddress: Clone + Sync + Send + FromStr + 'static,
 > {
-	/// Genesis UTXO of the target Partner Chain
+	/// Genesis UTXO that uniquely identifies the target Partner Chain
 	#[arg(long)]
 	pub genesis_utxo: UtxoId,
-	/// Partner Chain address to be associated with the Cardano address
+	/// Partner Chain address to be associated with the Cardano stake address
 	#[arg(long, value_parser=parse_pc_address::<PartnerchainAddress>)]
 	pub partnerchain_address: PartnerchainAddress,
-	/// Cardano Stake Signing Key bytes in hex format. Its public key will be associated with partnerchain_address.
+	/// Ed25519 signing key for the Cardano stake address
```

Comment:
I prefer the previous message. It was more **helpful** than proposed alternative.
It is text visible in `--help` of the command, right?